pkg/networkutils/ippool.go

```diff
@@ -0,0 +1,60 @@
+package networkutils
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type IPPool []string
+
+func NewIPPool() IPPool {
+	return IPPool{}
+}
+
+func NewIPPoolFromString(fromString string) IPPool {
+	return IPPool(strings.Split(fromString, ","))
+}
+
+func NewIPPoolFromEnv(ipPoolEnvVar string) (IPPool, error) {
+	value, ok := os.LookupEnv(ipPoolEnvVar)
+	if !ok {
+		return NewIPPool(), fmt.Errorf("%s environment ip pool does not exist", ipPoolEnvVar)
+	}
+	if value != "" {
+		return NewIPPoolFromString(value), nil
+	}
+	return NewIPPool(), nil
+}
+
+func (ipPool *IPPool) ToString() string {
+	return strings.Join(*ipPool, ",")
+}
+
+func (ipPool *IPPool) IsEmpty() bool {
+	return len(*ipPool) == 0
+}
+
+func (ipPool *IPPool) AddIP(ip string) {
+	*ipPool = append(*ipPool, ip)
+}
+
+func (ipPool *IPPool) PopIP() (string, error) {
+	if ipPool.IsEmpty() {
+		return "", fmt.Errorf("ip pool is empty")
```

Comment:
nit: If the error msg doesn't require any formatting, use `errors.New` instead of `fmt.Errorf`
```suggestion
		return "", errors.New("ip pool is empty")
```